src/ray/raylet/node_manager.cc

```diff
@@ -3005,6 +3014,61 @@ void NodeManager::FinishAssignTask(const std::shared_ptr<Worker> &worker,
   }
 }
 
+void NodeManager::ProcessSubscribePlasma(
+    const std::shared_ptr<LocalClientConnection> &client, const uint8_t *message_data) {
+  std::shared_ptr<Worker> associated_worker = worker_pool_.GetRegisteredWorker(client);
+  if (associated_worker == nullptr) {
+    associated_worker = worker_pool_.GetRegisteredDriver(client);
+  }
+  if (associated_worker == nullptr) {
+    RAY_LOG(ERROR) << "No worker exists for CoreWorker with client: "
```

Comment:
I'm not fully sure how to handle this case. If this happens the core_worker will never hear back. I don't think there should be a case where a non-worker/non-driver subscribes, but I am not 100% sure.

Reply:
It could be the worker process crashed and cleaned up between rpc. very edge case. just log it is fine. It's your decision to use RAY_LOG(ERROR) or RAY_CHECK(associated_worker != nullptr). RAY_CHECK will exit the process with helpful messages.